networking_odl/common/client.py

```diff
@@ -60,3 +64,33 @@ def try_delete(self, urlpath):
             LOG.debug("%(urlpath)s doesn't exist", {'urlpath': urlpath})
             return False
         return True
+
+    def get(self, full_path, data=None):
+        return self.request('GET', full_path, data)
+
+    def request(self, method, full_path, data=None):
+        headers = {'Content-Type': 'application/json'}
+        url = self.root_url + full_path
+        LOG.debug("Sending METHOD (%(method)s) URL (%(url)s) JSON (%(data)s)",
+                  {'method': method, 'url': url, 'data': data})
+        response = requests.request(
+            method=method, url=url, headers=headers, data=data, auth=self.auth,
+            timeout=self.timeout)
+        response.raise_for_status()
+        return response
+
+    _root_url = None
+
+    @property
+    def root_url(self):
+        root_url = self._root_url
+        if root_url is None:
+            url = urlparse.urlparse(self.url)
+            port = ''
+            if url.port:
+                port = ':' + str(url.port)
+            else:
```

Comment:
This branch is redundant. Port has already being initialized as an empty string.